src/DeviceIdWidget/widget/DeviceIdWidget.xml

```diff
@@ -0,0 +1,68 @@
+    <properties>
+        <property key="deviceEntity" type="entity" required="true">
```

Comment:
This must be required because the device information needs to be saved to the database. if look at the code that commits information it does that against an entity.